src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -1452,11 +1479,25 @@ public void accept(CodeBuilder cb) {
                         }
                     }
 
+                    int coder  = JLA.stringInitCoder(),
+                        length = 0;
+                    if (staticConcat) {
+                        for (var constant : constants) {
+                            coder |= JLA.stringCoder(constant);
+                            length += constant.length();
+                        }
+                    }
+
                     /*
                      * coder = coder(this.coder, arg0, arg1, ... argN);
                      */
-                    cb.aload(thisSlot)
-                      .getfield(concatClass, "coder", CD_byte);
+                    if (staticConcat) {
+                        cb.ldc(coder);
```

Comment:
`coder` can only be 0 or 1, so using `loadConstant` to generate `iconst_0` or `iconst_1` is better. Other `ldc` with number can be potentially replaced by `loadConstant` to generate optimized instructions too.